### ISSUE TITLE
Exception when using transactionCoroutine extension method

### ISSUE THREAD

**Issue by sbuettner** [jOOQ/jOOQ#14133]
### Expected behavior

When using JOOQs coroutine extension method for Kotlin `DSLContext.transactionCoroutine` the expected beviour is a commited transaction without any exception when using an r2dbc connection pool.

### Actual behavior

The `.awaitFirstOrNull()` call used in `transactionCoroutine` seem to lead to exception in the r2dbc layer:
```
2022-10-25T16:44:09.210+02:00 DEBUG 49542 --- [actor-tcp-nio-1] io.r2dbc.postgresql.QUERY                : Executing query: delete from "public"."XXX"
2022-10-25T16:44:09.229+02:00 DEBUG 49542 --- [-1 @coroutine#2] io.r2dbc.postgresql.QUERY                : Executing query: ROLLBACK
2022-10-25T16:44:09.231+02:00 ERROR 49542 --- [-1 @coroutine#2] reactor.core.publisher.Operators         : Operator called default onErrorDropped

java.lang.IllegalStateException: The connection is closed
	at io.r2dbc.pool.PooledConnection.assertNotClosed(PooledConnection.java:208) ~[r2dbc-pool-1.0.0.RC1.jar:1.0.0.RC1]
	at io.r2dbc.pool.PooledConnection.commitTransaction(PooledConnection.java:116) ~[r2dbc-pool-1.0.0.RC1.jar:1.0.0.RC1]
	at io.r2dbc.pool.PooledConnection.commitTransaction(PooledConnection.java:44) ~[r2dbc-pool-1.0.0.RC1.jar:1.0.0.RC1]
	at org.jooq.impl.R2DBC$TransactionSubscription$1.lambda$onNext0$15(R2DBC.java:722) ~[jooq-3.17.4.jar:na]
	at org.jooq.impl.Internal$1.onComplete(Internal.java:464) ~[jooq-3.17.4.jar:na]
	at reactor.core.publisher.StrictSubscriber.onComplete(StrictSubscriber.java:123) ~[reactor-core-3.5.0-RC1.jar:3.5.0-RC1]
	at reactor.core.publisher.MonoCreate$DefaultMonoSink.success(MonoCreate.java:173) ~[reactor-core-3.5.0-RC1.jar:3.5.0-RC1]
	at kotlinx.coroutines.reactor.MonoCoroutine.onCompleted(Mono.kt:101) ~[kotlinx-coroutines-reactor-1.6.4.jar:na]
	at kotlinx.coroutines.AbstractCoroutine.onCompletionInternal(AbstractCoroutine.kt:93) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.JobSupport.tryFinalizeSimpleState(JobSupport.kt:294) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:856) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:828) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:100) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46) ~[kotlin-stdlib-1.7.20.jar:1.7.20-release-201(1.7.20)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
```

Changing the code by using the following statement instead of the extension method `transactionCoroutine`:
```
return transactionPublisher { tx ->
        mono { block(tx.dsl()) }
    }.awaitLast() as T
```

which is based on `.awaitLast()` provides the expected behaviour without any exceptions. 

### Steps to reproduce the problem

Just uncomment the code in line https://github.com/sbuettner/reactive-database-access-kotlin/blob/main/src/main/kotlin/sbuetter/demo/db/ReactiveDatabaseExtensions.kt#L13 and run the tests of the repository using:
```
docker-compose up
``` 
and 
```
./gradlew flywayMigrate test
```

### jOOQ Version

3.17.4

### Database product and version

Postgres 14.4

### Java Version

17

### OS Version

Mac Os X

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:r2dbc-postgresql:1.0.0.RC1

**Comment by lukaseder** [comment]
Thanks for your  report. This is probably related to https://github.com/jOOQ/jOOQ/issues/13802

**Comment by lukaseder** [comment]
#13802 was fixed in jOOQ 3.18.0 and 3.17.7. Can you confirm 3.17.7 fixes your problem?
